in_toto/models/metadata.py

```diff
@@ -168,7 +168,13 @@ def create_signature(self, signer: Signer) -> Signature:
         return super().sign(signer)
 
     def verify_signature(self, verification_key):
-        key = SSlibKey.from_securesystemslib_key(verification_key)
+        # Deepcopy to preserve `verification_key`, which might still be needed
+        # in calling context and would otherwise be destroyed in `from_dict`.
+        # NOTE: It would be nice to support `Key` natively in in-toto model.
+        key = Key.from_dict(
+            verification_key["keyid"], deepcopy(verification_key)
```

Comment:
> NOTE: It would be nice to support `Key` natively in in-toto model.

So verify_signature would accept `Key` rather than the dict? Which would mean we wouldn't have these deepcopy concerns because we wouldn't have `from_dict` at all, right?

Reply:
I'll submit a ticket for it.